src/main/scala/org/broadinstitute/dsde/firecloud/filematch/result/SuccessfulMatchResult.scala

```diff
@@ -16,4 +16,7 @@ case class SuccessfulMatchResult(firstFile: Path, secondFile: Path, id: String)
 object SuccessfulMatchResult {
   def fromStrings(firstFile: String, secondFile: String, id: String): SuccessfulMatchResult =
     SuccessfulMatchResult(new java.io.File(firstFile).toPath, new java.io.File(secondFile).toPath, id)
+
+  def apply(firstFile: Path, id: String): SuccessfulMatchResult =
+    SuccessfulMatchResult(firstFile, null, id)
```

Comment:
hmmm … intentionally using a null is a big no-no in Scala land. I'd rather this return a `PartialMatchResult` as Marc suggested, then have `FileMatcher` handle those partials better. I don't think this would be too bad:
* FileMatcher.findFirstFiles changes to return List[FileMatchResult] and includes both SuccessfulMatchResult and PartialMatchResult
* FileMatcher.performPairing only calls findSecondFiles on the successes and passes partials untouched

If you don't like using PartialMatchResult in this way, you could also add another FileMatchResult subclass which indicates "I found the first file but please don't even look for a second file"

Reply:
Done: https://github.com/broadinstitute/firecloud-orchestration/pull/1581/commits/bf414fcf7acfdedd9a1b42d76449d4b9c10677b2